File: ruby/packages/var-core/lib/oselvar/var/core/parse.rb
# frozen_string_literal: true

require 'oselvar/var/core/scanner'
require 'oselvar/var/core/structurer'

module Oselvar
  module Var
    module Core
      # Parse +source+ into a VarDoc: scan blocks, then group into Examples.
      # Port of parse.ts.
      module Parse
        module_function

        def parse(path, source, plugins = [])
          Structurer.structure(path, source, Scanner.scan(source, plugins))
        end
      end
    end
  end
end

File: python/packages/core/src/varar_core/parse.py
"""parse.py — port of typescript/packages/core/src/parse.ts.

Combines scan + structure into the top-level parse function.
"""

from __future__ import annotations

from varar_core.ast import Doc
from varar_core.scanner import scan
from varar_core.structurer import structure


def parse(path: str, source: str) -> Doc:
    """Parse *source* into a Doc: scan blocks then group them into Examples."""
    return structure(path, source, scan(source))
